src/Sylius/Bundle/ApiBundle/Serializer/CommandFieldItemIriToIdentifierDenormalizer.php

```diff
@@ -30,35 +31,64 @@ final class CommandFieldItemIriToIdentifierDenormalizer implements ContextAwareD
     /** @var CommandAwareInputDataTransformer */
     private $commandAwareInputDataTransformer;
 
-    /** @var CommandItemIriArgumentToIdentifierMapInterface */
-    private $commandItemIriArgumentToIdentifierMap;
-
     public function __construct(
         DenormalizerInterface $objectNormalizer,
         ItemIriToIdentifierConverterInterface $itemIriToIdentifierConverter,
-        CommandAwareInputDataTransformer $commandAwareInputDataTransformer,
-        CommandItemIriArgumentToIdentifierMapInterface $commandItemIriArgumentToIdentifierMap
+        CommandAwareInputDataTransformer $commandAwareInputDataTransformer
     ) {
         $this->objectNormalizer = $objectNormalizer;
         $this->itemIriToIdentifierConverter = $itemIriToIdentifierConverter;
         $this->commandAwareInputDataTransformer = $commandAwareInputDataTransformer;
-        $this->commandItemIriArgumentToIdentifierMap = $commandItemIriArgumentToIdentifierMap;
     }
 
     public function supportsDenormalization($data, $type, $format = null, array $context = [])
     {
-        return $this->commandItemIriArgumentToIdentifierMap->has($this->getInputClassName($context));
+        /** @psalm-var class-string $inputClassName */
+        $inputClassName = $this->getInputClassName($context);
+
+        /** @var array $classInterfaces */
+        $classInterfaces = class_implements($inputClassName);
+
+        foreach ($classInterfaces as $classInterface) {
+            if ($classInterface === CommandFieldItemIriToIdentifierAwareInterface::class) {
+                return true;
+            }
+        }
+
+        return false;
     }
 
     public function denormalize($data, $type, $format = null, array $context = [])
     {
         /** @psalm-var class-string $inputClassName */
         $inputClassName = $this->getInputClassName($context);
 
-        $fieldName = $this->commandItemIriArgumentToIdentifierMap->get($inputClassName);
-
-        if (array_key_exists($fieldName, $data)) {
-            $data[$fieldName] = $this->itemIriToIdentifierConverter->getIdentifier($data[$fieldName]);
+        /** @var array $classInterfaces */
+        $classInterfaces = class_implements($inputClassName);
+
+        foreach ($classInterfaces as $classInterface) {
+            if ($classInterface === CommandFieldItemIriToIdentifierAwareInterface::class) {
+                $classFields = get_class_vars($inputClassName);
+                foreach ($classFields as $classFieldName => $classFieldValue) {
+                    $serializedFieldName = null;
+
+                    if ($this->isConvertableField($classFieldName)) {
+                        $serializedFieldName = substr($classFieldName, 0, -4);
+                    }
+
+                    if (array_key_exists($serializedFieldName, $data)) {
```

Comment:
```suggestion
                    if (isset ($data[$serializedFieldName])) {
```
Or even "return early".